cctp_primitives/src/bit_vector/compression.rs

```diff
@@ -170,11 +185,20 @@ fn bzip2_compress(bit_vector: &[u8]) -> Result<Vec<u8>, Error> {
     Ok(bzip_compressed)
 }
 
-fn bzip2_decompress(compressed_bit_vector: &[u8]) -> Result<Vec<u8>, Error> {
-    let mut uncompressed_bitvector = Vec::new();
+fn bzip2_decompress(compressed_bit_vector: &[u8], max_decompressed_size: usize) -> Result<Vec<u8>, Error> {
+    let mut uncompressed_bitvector = Vec::with_capacity(max_decompressed_size);
     let mut decompressor = BzDecoder::new(compressed_bit_vector);
-    decompressor.read_to_end(&mut uncompressed_bitvector)?;
-    
+    let mut fixed_array = [0; DECOMPRESSION_CHUNK_SIZE];
+
+    loop {
```

Comment:
Maybe comment the rationale behind this piece of code

Reply:
Fixed.